Sources/Starknet/Network/BatchRequest.swift

```diff
@@ -0,0 +1,40 @@
+public class BatchRequest<U: Decodable, P: Encodable> {
+    let rpcPayloads: [JsonRpcPayload<P>]
+    let config: HttpNetworkProvider.Configuration
+    let networkProvider: HttpNetworkProvider
+
+    init(
+        rpcPayloads: [JsonRpcPayload<P>],
+        config: HttpNetworkProvider.Configuration,
+        networkProvider: HttpNetworkProvider
+    ) {
+        self.rpcPayloads = rpcPayloads
+        self.config = config
+        self.networkProvider = networkProvider
+    }
+
+    private func getOrderedRpcResults(rpcResponses: [JsonRpcResponse<U>]) -> [Result<U, StarknetProviderError>] {
+        var orderedRpcResults: [Result<U, StarknetProviderError>?] = Array(repeating: nil, count: rpcPayloads.count)
+        for rpcResponse in rpcResponses {
+            if let error = rpcResponse.error {
+                orderedRpcResults[rpcResponse.id] = .failure(StarknetProviderError.jsonRpcError(error.code, error.message, error.data))
+            } else if let result = rpcResponse.result {
+                orderedRpcResults[rpcResponse.id] = .success(result)
+            } else {
+                orderedRpcResults[rpcResponse.id] = .failure(StarknetProviderError.unknownError)
+            }
+        }
```

Comment:
this seems like it could be really slow for longer response lists. I suppose we need ordering to properly deserialize? is there any other way?

Reply:
We need to reorder the responses to match the order in which they were sent. This can be achieved with sorting `O(nlogn)` or direct placement `O(n)`. Second solution was accepted and implemented in JVM SDK, where there is same situation, [here](https://github.com/software-mansion/starknet-jvm/pull/455#discussion_r1595342132).